src/modules/cpu/kernel/erase.hpp

```diff
@@ -0,0 +1,322 @@
+
+        Rpp32u numBoxes = numBoxesTensor[batchCount];
+        RpptRoiLtrb *anchorBoxInfo = anchorBoxInfoTensor + batchCount * numBoxes;
+        T *colors = colorsTensor + batchCount * numBoxes * srcDescPtr->c;
+
+        T *srcPtrImage, *dstPtrImage;
+        srcPtrImage = srcPtr + batchCount * srcDescPtr->strides.nStride;
+        dstPtrImage = dstPtr + batchCount * dstDescPtr->strides.nStride;
+
+        T *srcPtrChannel, *dstPtrChannel;
+        srcPtrChannel = srcPtrImage + (roi.xywhROI.xy.y * srcDescPtr->strides.hStride) + (roi.xywhROI.xy.x * layoutParams.bufferMultiplier);
+        dstPtrChannel = dstPtrImage;
+        T userPixel3[3];
+
+        // Erase with fused output-layout toggle (NHWC -> NCHW)
+        if ((srcDescPtr->c == 3) && (srcDescPtr->layout == RpptLayout::NHWC) && (dstDescPtr->layout == RpptLayout::NCHW))
+        {
+            T *srcPtrRow, *dstPtrRowR, *dstPtrRowG, *dstPtrRowB;
+            srcPtrRow = srcPtrChannel;
+            dstPtrRowR = dstPtrChannel;
+            dstPtrRowG = dstPtrRowR + dstDescPtr->strides.cStride;
+            dstPtrRowB = dstPtrRowG + dstDescPtr->strides.cStride;
+
+            for(int i = 0; i < roi.xywhROI.roiHeight; i++)
+            {
+                T *srcPtrTemp, *dstPtrTempR, *dstPtrTempG, *dstPtrTempB;
+                srcPtrTemp = srcPtrRow;
+                dstPtrTempR = dstPtrRowR;
+                dstPtrTempG = dstPtrRowG;
+                dstPtrTempB = dstPtrRowB;
+                bool isErase = false;
+                Rpp32u bufferLength = 0;
+                T userPixelR, userPixelG, userPixelB;
+                int j;
+
+                for (j = 0; j < roi.xywhROI.roiWidth;)
+                {
+                    for(int count = 0; count < numBoxes; count++)
+                    {
+                        Rpp32u x1 = (Rpp32u) RPPPRANGECHECK(anchorBoxInfo[count].lt.x, roi.xywhROI.xy.x, roi.xywhROI.roiWidth);
+                        Rpp32u y1 = (Rpp32u) RPPPRANGECHECK(anchorBoxInfo[count].lt.y, roi.xywhROI.xy.y, roi.xywhROI.roiHeight);
+                        Rpp32u x2 = (Rpp32u) RPPPRANGECHECK(anchorBoxInfo[count].rb.x, x1, roi.xywhROI.roiWidth);
+                        Rpp32u y2 = (Rpp32u) RPPPRANGECHECK(anchorBoxInfo[count].rb.y, y1, roi.xywhROI.roiHeight);
+                        Rpp32u countMul3 = count * 3;
+                        userPixelR = colors[countMul3];
+                        userPixelG = colors[countMul3 + 1];
+                        userPixelB = colors[countMul3 + 2];
+                        if(i >= y1 && i <= y2 && j >= x1 && j <= x2)
+                        {
+                            isErase = true;
+                            bufferLength = x2 - x1 + 1;
+                            break;
+                        }
+                    }
+                    if(isErase && bufferLength)
+                    {
+                        memset(dstPtrTempR, userPixelR, bufferLength * sizeof(T));
```

Comment:
Please check and see if std::fill can be used here as mentioned in previous comment